src/tools/tor-print-ed-signing-cert.c

```diff
@@ -55,9 +56,26 @@ main(int argc, char **argv)
     return -5;
   }
 
-  time_t expires_at = (time_t)cert->exp_field * 60 * 60;
+  const time_t expiration = (const time_t)cert->exp_field * 60 * 60;
 
-  printf("Expires at: %s", ctime(&expires_at));
+  const struct tm *expires_at = localtime(&expiration);
+
+  char rfc822_str[16] = "";
+
+  setlocale(LC_TIME, "en_US_POSIX");
+
+/* Yes, we're fine with RFC822 being written in 1982 and not addressing Y2K. */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-y2k"
+  strftime(rfc822_str, sizeof(rfc822_str), "%a, %d %b %y %T %z", expires_at);
```

Comment:
strftime has a return value: checking it would have caught the truncation bug.
Let's check it anyway: it's good practice.